core/context-strategy/autoscale-context.ts

```diff
@@ -252,9 +264,17 @@ export class ConstantIntervalHeartbeatSyncStrategy implements HeartbeatSyncStrat
                 nextHeartbeatTime: nextHeartbeatTime,
                 syncState: HeartbeatSyncState.InSync,
                 syncRecoveryCount: 0, // sync recovery count = 0 means no recovery needed
-                seq: 1, // set to 1 because it is the first heartbeat
+                // use the device sequence if it exists or 1 as the initial sequence
+                seq: (!isNaN(deviceSyncInfo.sequence) && deviceSyncInfo.sequence) || 1,
```

Comment:
ternary pls, this one is probably fine but the ternary is clearer.

`!isNaN(deviceSyncInfo.sequence) ? deviceSyncInfo.sequence : 1`

Reply:
fixed